cpp/mg_utility/data_structures/graph_data.hpp

```diff
@@ -0,0 +1,34 @@
+/// @file graph_data.hpp
+/// This file contains the graph data - nodes, edges, variables.
+
+#pragma once
+
+namespace mg_graph {
+
+/// Node representation.
+///
+/// Node has id and type. Id must be unique in the graph.
+template <typename TSize> struct Node { TSize id; };
```

Comment:
I would give them a default value of `uint64_t`. Thoughts?
(`template <typename TSize = uint64_t>`)

Reply:
If defaulting to `uint64_t`, would the user be able to write something like: `mg_graph::Graph *graph;` (without explicitly writing the template parameter). If yes, I am definitely into the idea of setting the default value.

Reply:
Yep, that would be the point of it. Makes it easier for the most frequent use case and you allow template arg for specific stuff.